djgeojson/views.py

```diff
@@ -88,7 +88,32 @@ def get_queryset(self):
         """
         Inspired by Glen Roberton's django-geojson-tiles view
         """
-        self.z, self.x, self.y = map(int, self.args[:3])
+        try:
+            self.z, self.x, self.y = map(int, self.args[:3])
+        except AttributeError:
+            # let's try to get these as kwargs
+
+            self.z = self.kwargs.get('z', None)
+            self.x = self.kwargs.get('x', None)
+            self.y = self.kwargs.get('y', None)
+
+            if self.z is None:
+                raise ValueError('View parameter Z not provided in URL.')
+
+            if self.x is None:
+                raise ValueError('View parameter X not provided in URL.')
+
+            if self.y is None:
+                raise ValueError('View parameter Y not provided in URL.')
+
+            try:
+                self.z = int(self.z)
+                self.x = int(self.x)
+                self.y = int(self.y)
+            except ValueError:
+                # should we just reraise this?
+                raise ValueError('The values for view parameters Z, X or Y are not valid integers.')
```

Comment:
This will trigger a 500 error response, which is not what we want. It would be better to return a Bad Request response : https://docs.djangoproject.com/en/1.7/ref/request-response/#django.http.HttpResponseBadRequest